### PR TITLE
Baklava v2.0.0 redeploy and too much refactoring

### LINKED ISSUE #22: Redeploy to baklava after cutting a v2.0.0 release on mento-core

### PR DESCRIPTION
**Pull request by bowd**
### Description

Because we cut a v2.0.0 version on mento-core, I did another deployment to Baklava so everything would be in order.
During this deployment, I refactored some of the script structure here to help us and added some scripts to deal with reserve funding, so I could easily migrate funds between the previously deployed reserve and the new one.

### Other changes

I introduced a top-level `script/dev` folder which contains dev scripts that are used in conjecture with the deployment but aren't central to it. We already had the script to deploy the `MockBridgedUSDC` contract to testnets which didn't really feel like a core upgrade script. By changing the structure and naming I also added all `broadcast/dev-*` to gitignore so that it's clear that dev scripts shouldn't be recorded in our broadcast folders.

I also added a few more helper bash scripts to make life easier when running and testing out these deployment or dev scripts. Check the readme for details.

### Tested

Deployed to baklava

### Related issues

- Fixes #22 

### Backwards compatibility

N/A

### Documentation

N/A

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -16,20 +16,36 @@ fs_permissions = [
...
-# accidentally linking the wrong libraries on the wrong network.
+# The way to get around this is by running the script once, have it fail and then
+# specify the libraries under the specific network here
+#
+# However, it is important to note that the first time a deployment is run on a network
+# we will have to rerun a deployment after all libraries are deployed and linked here.
+# If not then the metadata of contracts deployed **before** the libraries were added,
+# contracts that do not require the libraries, will still have their bytecode different
+# because of the metadata hash, and the verification will fail. This is because
+# the metadata of a contract will contain the staticially linked libraries during compilation
```
Comment (philbow61) [inline foundry.toml:31]:
```suggestion
# the metadata of a contract will contain the statically linked libraries during compilation
```

File: README.md [hunk 2]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
-./bin/get_secrets.sh
+yarn secrets:get
 
 # Execute scripts using forge script command
 forge script DeployCircuitBreaker --rpc-url $BAKLAVA_RPC_URL --broadcast --legacy --verify --verifier sourcify
-
 ```
+
+## Deployment Structure
+
```
Comment (philbow61) [inline README.md:29]: I think its just script in the repo

File: README.md [hunk 3]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
+The deployment scripts are organized in the following structure:
+- `contracts/`: Contains helper contracts that aren't core, for example a DummyERC20 contract used on testnets
+- `scripts/`: Contains all the Foundry deployment scripts
```
Comment (philbow61) [inline README.md:32]:
```suggestion
- `script/`: Contains all the Foundry deployment scripts
```

File: README.md [hunk 4]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
+  - `scripts/upgrade/`: Contains all the upgrade scripts, which serve as a migration from one version of mento to another
```
Comment (philbow61) [inline README.md:33]:
```suggestion
  - `script/upgrades/`: Contains all the upgrade scripts, which serve as a migration from one version of mento to another 
```

File: README.md [hunk 5]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
-
 ```
+
+## Deployment Structure
+
+The deployment scripts are organized in the following structure:
+- `contracts/`: Contains helper contracts that aren't core, for example a DummyERC20 contract used on testnets
+- `scripts/`: Contains all the Foundry deployment scripts
+  - `scripts/upgrade/`: Contains all the upgrade scripts, which serve as a migration from one version of mento to another 
+    - `scripts/upgrade/MU01/`: Contains all the upgrade scripts to migrate from version v1.0.0. to v2.0.0
```
Comment (philbow61) [inline README.md:34]:
```suggestion
    - `script/upgrades/MU01/`: Contains all the upgrade scripts to migrate from version v1.0.0. to v2.0.0
```

File: README.md [hunk 6]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
+  - `scripts/dev/`: Contains dev scripts that are used in the deployment process, especially on testnets, but aren't central to the upgrade.
```
Comment (philbow61) [inline README.md:35]:
```suggestion
  - `script/dev/`: Contains dev scripts that are used in the deployment process, especially on testnets, but aren't central to the upgrade.
```

File: README.md [hunk 7]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
+  - `scripts/utils/`: Contains helpers and utilities used in the deployment process.
```
Comment (philbow61) [inline README.md:36]:
```suggestion
  - `script/utils/`: Contains helpers and utilities used in the deployment process.
```

**Review by philbow61 — APPROVED** [review]
LGTM 👍

File: README.md [hunk 8]
```diff
@@ -15,13 +15,80 @@ cd mento-deployment
...
+
+## Scripts
+
+The scripts tend to follow a similar structure, and are either simple helpers or wrappers for the `forge script` command.
+
+General options will include:
+- `-n`: The network to run on, e.g. `baklava` or `alfajores`
+- `-u`: The upgrade number, e.g. `MU01`
+
+Check the script file for more details on usage but here's a quick overview:
```
Comment (ninocomputer) [inline README.md:48]:
```suggestion
Check the script directory for more details on usage but here's a quick overview:
```

File: bin/clean.sh [hunk 9]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+##############################################################################
+# Script for cleaning the broadcast file for an upgrade + networ combo
```
Comment (ninocomputer) [inline bin/clean.sh:4]:
```suggestion
# Script for cleaning the broadcast file for an upgrade + network combo
```

**Review by ninocomputer — APPROVED** [review]

**Pull request merged**